test/jdk/java/foreign/TestLayouts.java

```diff
@@ -72,6 +76,8 @@ public void testNotEquals(MemoryLayout layout) {
     public void testTargetLayoutEquals() {
         MemoryLayout differentTargetLayout = ADDRESS.withTargetLayout(JAVA_CHAR);
         assertFalse(ADDRESS.equals(differentTargetLayout));
+        var equalButNotSame = ADDRESS.withTargetLayout(JAVA_INT).withTargetLayout(JAVA_CHAR);
+        assertFalse(ADDRESS.equals(equalButNotSame));
```

Comment:
I don't see what this extra code is testing that wasn't already tested by the lines above.

Was this intended as `assertTrue(differentTargetLayout .equals(equalButNotSame));` ?

Reply:
The idea was to create a layout that is equal but not the same so to touch `OfAddress::equals` at the end where the layouts are compared.

Reply:
Ok, I get that, but that's not what the test code seems to be doing. i.e.g the 2 layouts are not equal, AFAICS